aws/nat_gateway/main.tf

```diff
@@ -0,0 +1,32 @@
+# Create a single subnet for the NAT Gateway to live in
+# routed to the outside world
+module "nat-gw-subnets" {
```

Comment:
This creates completely new subnets, in addition to the ones [created here](https://github.com/tablexi/terraform_modules/pull/182/files#diff-bbf241e87d035b79721072d42b18c8cf86c0c6db27e61fdcb1cbddacda0b61d7R32), right?

Can we have just a single set of subnets, and pass them around (or use them directly) as appropriate?

Reply:
The reason I did it this way is because:
- When we create subnets (using the vpc/subnets module) it also creates routes. These routes must have a gateway identified.
- To create a nat-gateway, that gateway has to live in a subnet.

Let me think through what you're proposing:
- separate the subnet-creation from the route/route-table creation (by creating `vpc/routes`)
- change the eks module to use the `vpc/subnets` first, then create the NAT-Gateway, then use `vpc/routes`
- Update other users of the `vpc/subnets` module to use the new `vpc/routes` module.

Ok, that doesn't sound too bad, lemme try.

Reply:
@vandrijevik have a look at the update.

Reply:
I think what I am proposing is to just create the NAT gateway and appropriate routes in the existing `eks` module (so that you have access to the subnets right there in the file, along with other portions for the NAT gateway), before trying to extract it into a module, or move other pieces into smaller submodules 😄

Reply:
@rdunlop I’ve been trying to message you on Slack to discuss this, but Slack is down (for others too) 😞

I am not quite sure that `routes` as a module makes too much sense to me here — I think that ideally, using a NAT gateway should be an _option_ for an EKS cluster here, and not required (it is not something that we want, or need in all cases).

Can that be achieved by introducing a NAT Gateway module which can be layered optionally on top of the EKS module we have, via something like this being inserted around https://github.com/tablexi/infrastructure/blob/master/terraform/tablexi/eks.tf#L55?

```
module "staging-eks-nat-gateway" {
  source = "github.com/tablexi/terraform_modules?ref=v2020-12-23-13//aws/nat_gateway"

  subnets = module.staging-eks.subnets
}
```

The above snipped would pick one of the subnets passed in to use for placing the NAT gateway (the first one?), and set up the appropriate routes for all other subnets (which already have route tables and route table associations, so we would not even need to create them).

Reply:
I think that I've addressed your concern